python/sbp/client/loggers/json_logger.py

```diff
@@ -33,9 +33,9 @@ def fmt_msg(self, msg):
       data = self.dispatcher(msg).to_json_dict()
     except KeyError:
       data = msg.to_json_dict()
-    return {"delta": self.delta(),
-            "timestamp": self.timestamp(),
-            "data": data}
+    msg = {"delta": self.delta(), "timestamp": self.timestamp(), "data": data}
```

Comment:
Added a `metadata` field, and always populate it, even if empty. `tags` is a little arbitrary - `metadata` fits nicer with `data`. Potentially `header` could be used here, too.